### PR TITLE
Refactor Crypto module for JBallerina

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
Refactor native implementation and unit tests of Crypto module for JBallerina.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/DecodePrivateKey.java [hunk 1]
```diff
@@ -166,14 +94,12 @@ public static Object decodePrivateKey(Strand strand, Object keyStoreValue, Objec
                 privateKeyRecord.put(Constants.PRIVATE_KEY_RECORD_ALGORITHM_FIELD, privateKey.getAlgorithm());
                 return privateKeyRecord;
             } else {
-                return CryptoUtils.createCryptoError("not a valid RSA key");
+                return CryptoUtils.createCryptoError("ot a valid RSA key");
```
Comment (Shan1024) [inline stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/DecodePrivateKey.java]: Seems like accidentally edited.

File: stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/DecryptAesCbc.java [hunk 2]
```diff
@@ -38,18 +36,7 @@
...
-        byte[] key = ((BValueArray) keyBValue).getBytes();
-        BValue ivBValue = context.getRefArgument(2);
-        byte[] iv = null;
-        if (ivBValue != null) {
-            iv = ((BValueArray) ivBValue).getBytes();
-        }
-        String padding = context.getRefArgument(3).stringValue();
-        CryptoUtils.aesEncryptDecrypt(context, CryptoUtils.CipherMode.DECRYPT, Constants.CBC, padding, key, input, iv,
-                -1);
+
```
Comment (Shan1024) [inline stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/DecryptAesCbc.java]: Empty newline added to the end of most methods. I think this comes from formatting rule settings, but seems like the settings are incorrect.

File: stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/DecodePrivateKey.java (hunk 1 above)
Comment (ldclakmal) [reply stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/DecodePrivateKey.java]: Yeah. Thanks for pointing out.

File: stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/DecryptAesCbc.java (hunk 2 above)
Comment (ldclakmal) [reply stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/DecryptAesCbc.java]: Yes. I fixed that with https://github.com/ballerina-platform/ballerina-lang/pull/15909/commits/b37b79a0fd639cba691569f560c9f3ca40b7c8fd and the `execute` method is to be removed once we fully migrated to JBallerina.

File: stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/DecodePublicKey.java [hunk 3]
```diff
@@ -209,11 +118,11 @@ public static Object decodePublicKey(Strand strand, Object keyStoreValue, Object
...
             }
         } catch (FileNotFoundException e) {
-            throw new org.ballerinalang.jvm.util.exceptions.BallerinaException(
+            throw new BallerinaException(
                     "PKCS12 key store not found at: " + keyStoreFile.getAbsoluteFile());
         } catch (KeyStoreException | CertificateException | IOException e) {
-            throw new org.ballerinalang.jvm.util.exceptions.BallerinaException(
-                    "unable to open keystore: " + e.getMessage());
+            throw new BallerinaException(
+                    "Unable to open keystore: " + e.getMessage());
```
Comment (Shan1024) [inline stdlib/crypto/src/main/java/org/ballerinalang/stdlib/crypto/nativeimpl/DecodePublicKey.java]: Can be merged with the previous line.

File: stdlib/crypto/src/test/java/org/ballerinalang/stdlib/crypto/CryptoTest.java [hunk 4]
```diff
@@ -770,14 +768,14 @@ public void testEncryptRsaEcbPkcs1() {
...
+                new BString("ballerina"), new BString("ballerina"), new BString("PKCS1")};
         BValue[] returnValues = BRunUtil.invoke(compileResult, "testEncryptRsaEcb", args);
         Assert.assertFalse(returnValues == null || returnValues.length == 0 || returnValues[0] == null);
         Assert.assertTrue(returnValues[0] instanceof BValueArray);
 
         BValue[] args1 = {returnValues[0], new BString(confRoot.resolve("testKeystore.p12").toString()),
-                new BString("ballerina"), new BString("ballerina"), new BString("ballerina"), new BString("PKCS1"),
-                null};
+                new BString("ballerina"), new BString("ballerina"), new BString("ballerina"),
+                new BString("PKCS1")};
```
Comment (Shan1024) [inline stdlib/crypto/src/test/java/org/ballerinalang/stdlib/crypto/CryptoTest.java]: Seems like can be merged with the previous line.

File: stdlib/crypto/src/test/java/org/ballerinalang/stdlib/crypto/CryptoTest.java [hunk 5]
```diff
@@ -789,14 +787,15 @@ public void testEncryptRsaEcbOAEPwithMD5andMGF1() {
         BValueArray messageValue = new BValueArray(message);
 
         BValue[] args = {messageValue, new BString(confRoot.resolve("testKeystore.p12").toString()),
-                new BString("ballerina"), new BString("ballerina"), new BString("OAEPwithMD5andMGF1"), null};
+                new BString("ballerina"), new BString("ballerina"),
+                new BString("OAEPwithMD5andMGF1")};
```
Comment (Shan1024) [inline stdlib/crypto/src/test/java/org/ballerinalang/stdlib/crypto/CryptoTest.java]: Can be merged with the previous line.

File: stdlib/crypto/src/test/java/org/ballerinalang/stdlib/crypto/CryptoTest.java [hunk 6]
```diff
@@ -808,14 +807,15 @@ public void testEncryptRsaEcbOAEPWithSHA1AndMGF1() {
         BValueArray messageValue = new BValueArray(message);
 
         BValue[] args = {messageValue, new BString(confRoot.resolve("testKeystore.p12").toString()),
-                new BString("ballerina"), new BString("ballerina"), new BString("OAEPWithSHA1AndMGF1"), null};
+                new BString("ballerina"), new BString("ballerina"),
+                new BString("OAEPWithSHA1AndMGF1")};
```
Comment (Shan1024) [inline stdlib/crypto/src/test/java/org/ballerinalang/stdlib/crypto/CryptoTest.java]: Can be merged with the previous line.

File: stdlib/crypto/src/test/java/org/ballerinalang/stdlib/crypto/CryptoTest.java [hunk 7]
```diff
@@ -846,14 +847,15 @@ public void testEncryptRsaEcbOAEPwithSHA384andMGF1() {
         BValueArray messageValue = new BValueArray(message);
 
         BValue[] args = {messageValue, new BString(confRoot.resolve("testKeystore.p12").toString()),
-                new BString("ballerina"), new BString("ballerina"), new BString("OAEPwithSHA384andMGF1"), null};
+                new BString("ballerina"), new BString("ballerina"),
+                new BString("OAEPwithSHA384andMGF1")};
```
Comment (Shan1024) [inline stdlib/crypto/src/test/java/org/ballerinalang/stdlib/crypto/CryptoTest.java]: Can be merged with the previous line.

File: stdlib/crypto/src/test/java/org/ballerinalang/stdlib/crypto/CryptoTest.java [hunk 8]
```diff
@@ -865,14 +867,15 @@ public void testEncryptRsaEcbOAEPwithSHA512andMGF1() {
         BValueArray messageValue = new BValueArray(message);
 
         BValue[] args = {messageValue, new BString(confRoot.resolve("testKeystore.p12").toString()),
-                new BString("ballerina"), new BString("ballerina"), new BString("OAEPwithSHA512andMGF1"), null};
+                new BString("ballerina"), new BString("ballerina"),
+                new BString("OAEPwithSHA512andMGF1")};
```
Comment (Shan1024) [inline stdlib/crypto/src/test/java/org/ballerinalang/stdlib/crypto/CryptoTest.java]: Can be merged with the previous line.

File: stdlib/crypto/src/test/java/org/ballerinalang/stdlib/crypto/CryptoTest.java [hunk 9]
```diff
@@ -884,15 +887,14 @@ public void testEncryptRsaEcbWithPrivateKeyPkcs1() {
         BValueArray messageValue = new BValueArray(message);
 
         BValue[] args = {messageValue, new BString(confRoot.resolve("testKeystore.p12").toString()),
-                new BString("ballerina"), new BString("ballerina"), new BString("ballerina"), new BString("PKCS1"),
-                null};
+                new BString("ballerina"), new BString("ballerina"), new BString("ballerina"),
+                new BString("PKCS1")};
```
Comment (Shan1024) [inline stdlib/crypto/src/test/java/org/ballerinalang/stdlib/crypto/CryptoTest.java]: Seems like can be merged with the previous line.

**Pull request merged**